### PR TITLE
1 como cliente quiero recibir información a cerca del estado de una criptomoneda

### LINKED ISSUE #1: Como cliente quiero recibir información a cerca del estado de una criptomoneda
**Descripción**
Recibo una petición GET y devuelvo el estado de la criptomoneda

**Ejemplo**

ID de la criptomoneda no existe:
Request: GET /coin/status/coin_id

Response: 404 Not Found
```
{ 
    error : A coin with the specified ID was not found
}
```

Servicio no disponible:
Request: GET /coin/status/coin_id

Response: 503 Service Unavailable
```
{ 
    error : Service unavailable
}
```

Happy path:
Request: GET /coin/status/coin_id

Response: 200 OK
```
{
    coin_id: string,
    symbol: string,
    name: string,
    name_id: string,
    rank: integer,
    price_usd: string
}
```

### PR DESCRIPTION
**Pull request by xAriextz**
Pull request for issue #1

### CODE REVIEW

File: app/Application/CryptoDataSource/CryptoDataSource.php [hunk 1]
```diff
@@ -0,0 +1,10 @@
+<?php
+
+namespace App\Application\CryptoDataSource;
+
+use App\Domain\CoinStatus;
+
+interface CryptoDataSource
+{
+    public function findCoinById(int $coinId): CoinStatus;
```
Comment (jesuslezaun) [inline app/Application/CryptoDataSource/CryptoDataSource.php]: rename by findCoinStatusById

Comment (jesuslezaun) [inline app/Application/CryptoDataSource/CryptoDataSource.php]: coinId should be string

File: tests/app/Application/CoinStatus/CoinStatusServiceTest.php [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+    {
+        $coinStatus = new CoinStatus("90", "BTC", "Bitcoin", "bitcoin", 1, "600");
+
+        $this->cryptoDataSource
+            ->expects('findCoinById')
+            ->with("90")
+            ->once()
+            ->andReturn($coinStatus);
+
+        $coinStatusService = $this->coinStatusService->getCoinStatus("90");
```
Comment (jesuslezaun) [inline tests/app/Application/CoinStatus/CoinStatusServiceTest.php]: Change name to responseCoinStatus

File: tests/app/Application/CoinStatus/CoinStatusServiceTest.php [hunk 3]
```diff
@@ -0,0 +1,83 @@
+<?php
+
+namespace Tests\app\Application\CoinStatus;
```
Comment (raulsc-cl) [inline tests/app/Application/CoinStatus/CoinStatusServiceTest.php]: Use uppercase for "app"

File: tests/app/Infrastructure/Controller/CoinStatusControllerTest.php [hunk 4]
```diff
@@ -0,0 +1,84 @@
+<?php
+
+namespace Tests\app\Infrastructure\Controller;
```
Comment (raulsc-cl) [inline tests/app/Infrastructure/Controller/CoinStatusControllerTest.php]: Use uppercase for "app"

**Review by jesuslezaun — CHANGES_REQUESTED** [review]
Some small changes to check

**Review by raulsc-cl — CHANGES_REQUESTED** [review]
Some changes to be checked

**Review by jesuslezaun — APPROVED** [review]
Ok to merge

**Review by raulsc-cl — APPROVED** [review]
Okay to merge

**Pull request merged**